### PR TITLE
ci: binary size included in each release

### PR DESCRIPTION
**Pull request by levibostian**
Part of: https://linear.app/customerio/issue/MBL-155/track-the-binary-size-of-our-ios-sdk-upon-every-release

Allowing us to see the binary size of each release of our SDK. This is done by generating binary size reports to local files. Then, these files are committed to git on each release.

Testing criteria:
To test this, I modified the deploy-sdk workflow to only generate the report files (skip running semantic-release and cocoapods). I then ran this workflow by creating a draft PR.

Notes for reviewers:
* binny is complex to get working on Linux. I noticed that you either have to compile it from source or use outdated pre-made builds of it. Installing on mac is more sustainable long-term so I suggest we stick to running bloaty on mac, only, on the CI.
* I noticed that I was beginning to copy and paste the same code over and over again for downloading the latest main branch build and generating the SDK size reports. For this commit, I opted to making these tasks re-usable by creating dedicated github actions to do these repetitive tasks. In a future PR, I plan to refactor the workflow files to all use these re-usable actions. That's out of scope for this commit.

---

**Stack**:
- #699
- #697
- #696 ⬅
- #695
- #694

⚠️ *Part of a stack created by [spr](https://github.com/ejoffe/spr). Do not merge manually using the UI - doing so may have unexpected results.*

### CODE REVIEW

File: .github/actions/main-sample-app-build/action.yml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    value: ${{ steps.set-action-outputs.outputs.apn-app-xcarchive-name }}
+
+runs:
+  using: "composite"
+  steps:
+    - name: Setting environment variables for convenience in future steps 
+      shell: bash
+      # Use each of the set variables in future steps like: ${{ env.APN_SAMPLE_APP_BUILD_PATH }}      
+      run: |
+        echo "APN_SAMPLE_APP_BUILD_PATH=Apps/APN-UIKit/build" >> $GITHUB_ENV
```
Comment (ami-aman) [inline .github/actions/main-sample-app-build/action.yml:27]: What are the possible values of `GITHUB_ENV` ?

**Review by ami-aman — APPROVED** [review]
lgtm

File: .github/actions/main-sample-app-build/action.yml (hunk 1 above)
Comment (levibostian) [reply .github/actions/main-sample-app-build/action.yml:27]: GITHUB_ENV contains all environment variables of the CI job. So, the possibilities are infinite! 

The syntax of this line of code, `echo "FOO=BAR" >> $GITHUB_ENV`, is how you set new environment variables on github actions. I understand that might not be very clear looking at it.

**Comment by levibostian** [comment]
✓ Commit merged in pull request [#697](https://github.com/customerio/customerio-ios/pull/697)

**Pull request closed without merging**